<?xml version="1.0" encoding="UTF-8"?>
<testng-results retried="1" ignored="1" total="7" passed="4" failed="1" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2024-04-27T16:06:30 IST" name="Suite" finished-at="2024-04-27T16:07:13 IST" duration-ms="42372">
    <groups>
      <group name="ErrorHandling">
        <method signature="ErrorValidations.getErrorMessage()[pri:0, instance:rahulshettyacadamy.tests.ErrorValidations@3e44f2a5]" name="getErrorMessage" class="rahulshettyacadamy.tests.ErrorValidations"/>
      </group> <!-- ErrorHandling -->
      <group name="purchase">
        <method signature="SubmitOrderTest.SubmitOrder(java.util.HashMap)[pri:0, instance:rahulshettyacadamy.tests.SubmitOrderTest@62656be4]" name="SubmitOrder" class="rahulshettyacadamy.tests.SubmitOrderTest"/>
      </group> <!-- purchase -->
    </groups>
    <test started-at="2024-04-27T16:06:30 IST" name="Submit Order Test" finished-at="2024-04-27T16:07:04 IST" duration-ms="33710">
      <class name="rahulshettyacadamy.tests.SubmitOrderTest">
        <test-method is-config="true" signature="launchApplication()[pri:0, instance:rahulshettyacadamy.tests.SubmitOrderTest@62656be4]" started-at="2024-04-27T16:06:31 IST" name="launchApplication" finished-at="2024-04-27T16:06:34 IST" duration-ms="3090" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchApplication -->
        <test-method signature="SubmitOrder(java.util.HashMap)[pri:0, instance:rahulshettyacadamy.tests.SubmitOrderTest@62656be4]" started-at="2024-04-27T16:06:34 IST" name="SubmitOrder" data-provider="getData" finished-at="2024-04-27T16:06:53 IST" duration-ms="19145" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[{password=Admin@123, product=ZARA COAT 3, email=qa@mail.com}]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SubmitOrder -->
        <test-method is-config="true" signature="teardown()[pri:0, instance:rahulshettyacadamy.tests.SubmitOrderTest@62656be4]" started-at="2024-04-27T16:06:53 IST" name="teardown" finished-at="2024-04-27T16:06:53 IST" duration-ms="260" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
        <test-method is-config="true" signature="launchApplication()[pri:0, instance:rahulshettyacadamy.tests.SubmitOrderTest@62656be4]" started-at="2024-04-27T16:06:53 IST" name="launchApplication" finished-at="2024-04-27T16:06:58 IST" duration-ms="4221" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchApplication -->
        <test-method signature="SubmitOrder(java.util.HashMap)[pri:0, instance:rahulshettyacadamy.tests.SubmitOrderTest@62656be4]" started-at="2024-04-27T16:06:58 IST" name="SubmitOrder" data-provider="getData" finished-at="2024-04-27T16:07:01 IST" duration-ms="3580" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[{password=Admin@123, product=ADIDAS ORIGINAL, email=error@gmail.com}]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SubmitOrder -->
        <test-method is-config="true" signature="teardown()[pri:0, instance:rahulshettyacadamy.tests.SubmitOrderTest@62656be4]" started-at="2024-04-27T16:07:01 IST" name="teardown" finished-at="2024-04-27T16:07:01 IST" duration-ms="181" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
        <test-method is-config="true" signature="launchApplication()[pri:0, instance:rahulshettyacadamy.tests.SubmitOrderTest@62656be4]" started-at="2024-04-27T16:07:01 IST" name="launchApplication" finished-at="2024-04-27T16:07:03 IST" duration-ms="1530" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchApplication -->
        <test-method signature="orderHistoryTest()[pri:0, instance:rahulshettyacadamy.tests.SubmitOrderTest@62656be4]" started-at="2024-04-27T16:07:03 IST" name="orderHistoryTest" finished-at="2024-04-27T16:07:04 IST" depends-on-methods="rahulshettyacadamy.tests.SubmitOrderTest.SubmitOrder" duration-ms="1008" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- orderHistoryTest -->
        <test-method is-config="true" signature="teardown()[pri:0, instance:rahulshettyacadamy.tests.SubmitOrderTest@62656be4]" started-at="2024-04-27T16:07:04 IST" name="teardown" finished-at="2024-04-27T16:07:04 IST" duration-ms="193" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
      </class> <!-- rahulshettyacadamy.tests.SubmitOrderTest -->
    </test> <!-- Submit Order Test -->
    <test started-at="2024-04-27T16:06:30 IST" name="Error Validation Test" finished-at="2024-04-27T16:07:13 IST" duration-ms="42372">
      <class name="rahulshettyacadamy.tests.ErrorValidations">
        <test-method is-config="true" signature="launchApplication()[pri:0, instance:rahulshettyacadamy.tests.ErrorValidations@3e44f2a5]" started-at="2024-04-27T16:06:30 IST" name="launchApplication" finished-at="2024-04-27T16:06:38 IST" duration-ms="7165" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchApplication -->
        <test-method retried="true" signature="getErrorMessage()[pri:0, instance:rahulshettyacadamy.tests.ErrorValidations@3e44f2a5]" started-at="2024-04-27T16:06:38 IST" name="getErrorMessage" finished-at="2024-04-27T16:06:39 IST" duration-ms="908" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [Incorrect email or password.] but found [Incorrect email r password.]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [Incorrect email or password.] but found [Incorrect email r password.]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertEqualsImpl(Assert.java:150)
at org.testng.Assert.assertEquals(Assert.java:132)
at org.testng.Assert.assertEquals(Assert.java:656)
at org.testng.Assert.assertEquals(Assert.java:666)
at rahulshettyacadamy.tests.ErrorValidations.getErrorMessage(ErrorValidations.java:27)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:228)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:961)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:201)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:819)
at org.testng.TestRunner.run(TestRunner.java:619)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getErrorMessage -->
        <test-method is-config="true" signature="teardown()[pri:0, instance:rahulshettyacadamy.tests.ErrorValidations@3e44f2a5]" started-at="2024-04-27T16:06:39 IST" name="teardown" finished-at="2024-04-27T16:06:39 IST" duration-ms="269" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
        <test-method is-config="true" signature="launchApplication()[pri:0, instance:rahulshettyacadamy.tests.ErrorValidations@3e44f2a5]" started-at="2024-04-27T16:06:39 IST" name="launchApplication" finished-at="2024-04-27T16:06:42 IST" duration-ms="2769" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchApplication -->
        <test-method signature="getErrorMessage()[pri:0, instance:rahulshettyacadamy.tests.ErrorValidations@3e44f2a5]" started-at="2024-04-27T16:06:42 IST" name="getErrorMessage" finished-at="2024-04-27T16:06:43 IST" duration-ms="1315" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [Incorrect email or password.] but found [Incorrect email r password.]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [Incorrect email or password.] but found [Incorrect email r password.]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertEqualsImpl(Assert.java:150)
at org.testng.Assert.assertEquals(Assert.java:132)
at org.testng.Assert.assertEquals(Assert.java:656)
at org.testng.Assert.assertEquals(Assert.java:666)
at rahulshettyacadamy.tests.ErrorValidations.getErrorMessage(ErrorValidations.java:27)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.retryFailed(TestInvoker.java:249)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:76)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:961)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:201)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:819)
at org.testng.TestRunner.run(TestRunner.java:619)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:481)
at org.testng.internal.thread.ThreadUtil.lambda$execute$0(ThreadUtil.java:58)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getErrorMessage -->
        <test-method is-config="true" signature="teardown()[pri:0, instance:rahulshettyacadamy.tests.ErrorValidations@3e44f2a5]" started-at="2024-04-27T16:06:44 IST" name="teardown" finished-at="2024-04-27T16:06:44 IST" duration-ms="196" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
        <test-method is-config="true" signature="launchApplication()[pri:0, instance:rahulshettyacadamy.tests.ErrorValidations@3e44f2a5]" started-at="2024-04-27T16:06:44 IST" name="launchApplication" finished-at="2024-04-27T16:06:48 IST" duration-ms="4669" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchApplication -->
        <test-method signature="getProductErrorMessage()[pri:0, instance:rahulshettyacadamy.tests.ErrorValidations@3e44f2a5]" started-at="2024-04-27T16:06:48 IST" name="getProductErrorMessage" finished-at="2024-04-27T16:07:12 IST" duration-ms="24067" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- getProductErrorMessage -->
        <test-method is-config="true" signature="teardown()[pri:0, instance:rahulshettyacadamy.tests.ErrorValidations@3e44f2a5]" started-at="2024-04-27T16:07:12 IST" name="teardown" finished-at="2024-04-27T16:07:13 IST" duration-ms="344" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
      </class> <!-- rahulshettyacadamy.tests.ErrorValidations -->
    </test> <!-- Error Validation Test -->
  </suite> <!-- Suite -->
</testng-results>
